activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,28 @@
+*   Add expiry and purpose to MessageEncryptor & MessageVerifier.
+    
+    Messages encrypted and generated using MessageEncryptor and Message Verifier now
+    ActiveSupport expiry and purpose. You can set the purpose of the message using the key
+    :purpose, and likewise, the expiration using :expires_at or :expires_in.
+
+    This ensures that the message is used only for its intended purpose and also while it
+    hasn't expired.
+
+    Usage example: Message Encryptor
+
+        token = @encryptor.encrypt_and_sign(address, purpose: "shipping", expires_at: Time.now.end_of_year)
+        @encryptor.decrypt_and_verify(token, purpose: "shipping")
+
+    Usage example: Message Verifier
+
+        token = @verifier.generate(address, purpose: "shipping", expires_in: 1.month)
+        @verifier.verified(token, purpose: "shipping")
+        @verifier.verify(token, purpose: :shipping)
+
+    #29599 - Added expiry and purpose to MessageEncryptor
+    #29854 - Added expiry and purpose to MessageVerifier
```

Comment:
Let's just put these down like `Pull requests: #29599, #29854` (add this one too), as people can see the title when they click on it.